src/skmultiflow/data/base_generator.py

```diff
@@ -283,20 +283,20 @@ def has_more_samples(self):
 
     def get_data_info(self):
         """ get_name
-        
-        Gets the name of the plot, which is a string that will appear 
+
+        Gets the name of the plot, which is a string that will appear
         in evaluation methods, to represent the stream.
-        
+
         The default format is: 'Stream name - x labels'.
-        
+
         Returns
         -------
         string
             A string representing the plot name.
-        
+
         """
         return self.name + " - {} target(s), {} classes, {} features".format(self.n_targets,
                                                                              self.n_classes, self.n_features)
 
     def get_class_type(self):
-        return 'stream'
+        return type(self)
```

Comment:
Always a multiflow-specific type here? Streamz comes in many classes, hope it's not possible to end up here with one of those.

Reply:
`get_class_type` has been deprecated. However, we could take the chance to define a method in the base class to return the stream (streamz) type.

Reply:
> However, we could take the chance to define a method in the base class to return the stream (streamz) type

Don't think so since this PR tries to separate generators from streams